fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -409,7 +411,8 @@ public Response createOrReplaceObjectRdf(
             //}
         }
 
-        final String fedoraId = identifierConverter().toInternalId(externalPath());
+        final String externalUri = this.uriInfo.getRequestUri().toString();
+        final String fedoraId = identifierConverter().toInternalId(externalUri);
```

Comment:
Just noting that we get the `externalPath` from the [`@PathParam`](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java#L138) so adding the host and context back on is why I added the [`toDomain()`](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java#L169) method to the HttpIdentifierConverter. So these two lines could be
```
final String fedoraId = identifierConverter().toInternalId(identifierConverter().toDomain(externalPath()));
```
Otherwise we could remove the `externalPath()` setup and make either `externalUri` and/or `fedoraId` into a class variable as this will probably be needed.

Reply:
To be addressed post PR.

Reply:
I'll updated it according to your suggestion.

Reply:
@whikloj :  I'd prefer to defer this change for another PR in the interest of moving things along.  Would you mind creating a JIRA describing the changes you are recommending?

Reply:
https://jira.lyrasis.org/browse/FCREPO-3170